packer/rpc/communicator.go

```diff
@@ -115,7 +132,12 @@ func (c *communicator) Start(cmd *packer.RemoteCmd) (err error) {
 func (c *communicator) Upload(path string, r io.Reader, fi *os.FileInfo) (err error) {
 	// Pipe the reader through to the connection
 	streamId := c.mux.NextId()
-	go serveSingleCopy("uploadData", c.mux, streamId, nil, r)
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
👍 I don't have a strong preference here but for wg(1) you could also use a channel. E.g. 
```
done := make(chan struct{})
go func() {
    close(done)
}
<-done
```